students/adam_strong/lesson03/assignment/basic_operations.py

```diff
@@ -0,0 +1,116 @@
+                                        home_address=customer_tuple[HOME_ADDRESS],
+                                        phone_number=customer_tuple[PHONE_NUMBER],
+                                        email_address=customer_tuple[EMAIL_ADDRESS],
+                                        is_active=customer_tuple[IS_ACTIVE],
+                                        credit_limit=customer_tuple[CREDIT_LIMIT])
+
+        new_customer.save()
+        logger.info('Adding new customer: %s', str(new_customer))
+        logger.info('Added %s %s to database', new_customer.first_name, new_customer.last_name)
+        return model_to_dict(new_customer)
+
+
+def search_customer(customer_id):
+    ''' This function will return customer info based on a customer_id.'''
+    logger.info('Searching for customer #%s', str(customer_id))
+
+    with database.transaction():
+        try:
+            c = Customers.get(Customers.customer_id == customer_id)
+            logger.info(model_to_dict(c))
+            return {'first_name': c.first_name, 'last_name': c.last_name,
+                    'email_address': c.email_address, 'phone_number': c.phone_number}
+        except (IndexError, peewee.DoesNotExist):
+            return {}
+
+
+def delete_customer(customer_id):
+    '''This function will delete a customer from the sqlite3 database.'''
+    logger.info('Deleting customer #%s', str(customer_id))
+
+    with database.transaction():
+        c = Customers.get(Customers.customer_id == customer_id)
+        logger.info('Deleting %s %s from the database', c.first_name, c.last_name)
+        c.delete_instance()
+
+
+
+def update_customer(customer_id, credit_limit):
+    '''This function will search an existing customer by customer_id and update
+    their credit limit or raise a ValueError exception if the customer does not exist.'''
+    logger.info('Modifying customer #%s credit limit to %s', customer_id, credit_limit)
+
+    with database.transaction():
+        try:
+            c = Customers.get(Customers.customer_id == customer_id)
+            logger.info('%s %slimit is now %s', c.first_name, c.last_name, credit_limit)
+            c.credit_limit = credit_limit
+            c.save()
+            return {'first_name': c.first_name, 'last_name': c.last_name,
+                    'email_address': c.email_address, 'credit_limit': c.credit_limit}
+        except:
+            raise ValueError('No customer with this ID, use list_all_customers()')
+
+
+def list_active_customers():
+    '''This function will return an integer with the current number of active customers.'''
+    query = Customers.select(fn.COUNT(Customers.customer_id)).where(Customers.is_active == 1)
+    count = query.scalar()
+    logger.info('There are a total of %s active customers', str(count))
+    return count
+
+
+def list_all_customers():
+    '''This function returns a list of lists of all the customers'''
+    cust_list = []
+    for c in Customers:
```

Comment:
Remember _Customers_ is a SQL db table, so you could get a list of all customers by running a query::

`Customers.select()`

If you only wanted active customers:

`Customers.select().where(Customers.status == "active")`